rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/InterpretedLanguageContributions.java

```diff
@@ -256,19 +267,35 @@ private CompletableFuture<IConstructor> parseCommand(String command) {
         });
     }
 
-    private static CompletableFuture<@Nullable IFunction> getFunctionFor(CompletableFuture<ISet> contributions, String cons) {
+    private static CompletableFuture<@Nullable IConstructor> getContribution(CompletableFuture<ISet> contributions, String cons) {
         return contributions.thenApply(conts -> {
             for (IValue elem : conts) {
                 IConstructor contrib = (IConstructor) elem;
                 if (cons.equals(contrib.getConstructorType().getName())) {
-                    return (IFunction) contrib.get(0);
+                    return contrib;
                 }
             }
             logger.debug("No {} defined", cons);
             return null;
         });
     }
 
+    private static CompletableFuture<@Nullable IFunction> getFunctionFor(CompletableFuture<ISet> contributions, String cons) {
+        return getContribution(contributions, cons).thenApply(contribution -> (IFunction) contribution.get(0));
+    }
+
+    private static CompletableFuture<@Nullable IFunction> getRenameFunction(CompletableFuture<ISet> contributions) {
```

Comment:
why is this the only function that gets a special overload? and the other use `getFunctionFor`? should this maybe be an overload called `getParameterizedFunction`? or something like this? (the discrepancy is more clear in the constructor)

Reply:
I have now unified the "getFunctionFor" approach.